src/main/java/at/hannibal2/skyhanni/features/event/spook/TheGreatSpook.kt

```diff
@@ -1,84 +1,170 @@
 import at.hannibal2.skyhanni.utils.LorenzUtils
 import at.hannibal2.skyhanni.utils.NEUCalculator
 import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
 import at.hannibal2.skyhanni.utils.RenderUtils.renderString
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
 import at.hannibal2.skyhanni.utils.SoundUtils
 import at.hannibal2.skyhanni.utils.StringUtils
-import at.hannibal2.skyhanni.utils.TabListData
+import at.hannibal2.skyhanni.utils.TimeUnit
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.renderables.Renderable
 import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
 
 @SkyHanniModule
 object TheGreatSpook {
-
-    // §r§cPrimal Fears§r§7: §r§6§lREADY!!
     private val config get() = SkyHanniMod.feature.event.spook
-    private var displayTimer = ""
-    private var displayTimeLeft = ""
-    private var notificationSeconds = 0
+
+    private var isGreatSpookActive = false
+    private var greatSpookTimeRange: LongRange? = null
+    private var greatSpookEnd = SimpleTimeMark.farPast()
+
+    private var displayMobCooldown: Renderable? = null
+    private var displayGreatSpookEnd: Renderable? = null
+
+    private var timeUntilNextMob = SimpleTimeMark.farPast()
+
+    private val patternGroup = RepoPattern.group("event.greatspook")
+    /**
+     * REGEX-TEST: §d§lQUICK MATHS! §r§7Solve: §r§e(10*2)+12*5
+     */
+    private val mathFearMessagePattern by patternGroup.pattern(
+        "chat.math",
+        "§d§lQUICK MATHS! §r§7Solve: §r§e(?<math>.*)",
+    )
+    /**
+     * REGEX-TEST: §4[FEAR] Public Speaking Demon§r§f: Speak PlasticEating!
+     */
+    private val speakingFearMessagePattern by patternGroup.pattern(
+        "chat.speaking",
+        "§4\\[FEAR] Public Speaking Demon§r§f: (Speak|Say something interesting) (?<name>.*)!",
+    )
+    /**
+     * REGEX-TEST: §5§lFEAR. §r§eA §r§dPrimal Fear §r§ehas been summoned!
+     */
+    private val primalFearSpawnPattern by patternGroup.pattern(
+        "mob.spawn",
+        "§5§lFEAR\\. §r§eA §r§dPrimal Fear §r§ehas been summoned!",
+    )
+
+    private var noFearMessageCooldown = SimpleTimeMark.farPast()
 
     @SubscribeEvent
     fun onSecondPassed(event: SecondPassedEvent) {
         if (!LorenzUtils.inSkyBlock) return
+        if (!isGreatSpookActive) return
+
+        val fear = SkyblockStat.FEAR.lastKnownValue ?: run {
+            if (noFearMessageCooldown.isInPast() && (config.primalFearNotification || config.primalFearTimer))
+                noFearMessageCooldown = SimpleTimeMark.now().plus(2.minutes)
+            ChatUtils.userError("Fear stat not found! Please enable the Stats widget and enable the Fear stat for the best results.")
```

Comment:
I feel like this message might be better if it only shows when the player has spawned a fear mob